lib/led/rgb.js

```diff
@@ -275,22 +281,16 @@ class RGB {
         blue: 255
       };
 
-      state.prev = null;
-
       this.update(colors);
     }
 
     return this;
   }
 
   off() {
-    const state = priv.get(this);
-
-    // If it's already off, do nothing so the pervious state stays intact
+    // If it's already off, do nothing so the previous state stays intact
     /* istanbul ignore else */
     if (this.isOn) {
-      state.prev = RGB.colors.reduce((current, color) => (current[color] = state[color], current), {});
-
```

Comment:
Within the `off()` method, there is no need to reassign `state.prev` if the RGB is on since `state.prev` is assigned when the `on()` method is called. This also ensures that if `off()` is called when the RGB is pulsing that the RGB values are NOT assigned lesser and lesser values until reaching (0,0,0) which is the case when `off()` is repeatedly called after pulse() is called if this line remains.